backend/src/test/java/botobo/core/domain/workbook/WorkbookSearchRepositoryTest.java

```diff
@@ -0,0 +1,320 @@
+                .searchKeyword("Javascript")
+                .searchCriteria("name")
+                .build();
+
+        // when
+        Page<Workbook> workbooks = workbookSearchRepository.searchAll(parameter, null, null, parameter.toPageRequest());
+        List<Workbook> workbookList = workbooks.toList();
+
+        // then
+        assertThat(workbookList).hasSize(3);
+        assertThat(workbookList.stream()
+                .map(Workbook::getName)
+                .collect(Collectors.toList())).containsExactly("Javascript 문제집0",
+                "Javascript 문제집1", "Javascript 문제집2");
+    }
+
+    @Test
+    @DisplayName("검색어를 입력하고 카드순으로 정렬한다.")
+    void searchAllFromKeywordAndCardDesc() {
+        // given
+        WorkbookSearchParameter parameter = WorkbookSearchParameter.builder()
+                .searchKeyword("Javascript")
+                .searchCriteria("count")
+                .build();
+
+        // when
+        Page<Workbook> workbooks = workbookSearchRepository.searchAll(parameter, null, null, parameter.toPageRequest());
+        List<Workbook> workbookList = workbooks.toList();
+
+        // then
+        assertThat(workbookList).hasSize(3);
+        assertThat(workbookList.stream()
+                .map(Workbook::getName)
+                .collect(Collectors.toList())).containsExactly("Javascript 문제집0",
+                "Javascript 문제집1", "Javascript 문제집2");
+    }
+
+    @Test
+    @DisplayName("검색어를 입력하고 좋아요순으로 정렬한다. 좋아요가 같다면 id순으로 정렬한다.")
+    void searchAllFromKeywordAndHeartDesc() {
+        // given
+        WorkbookSearchParameter parameter = WorkbookSearchParameter.builder()
+                .searchKeyword("문제")
+                .searchCriteria("heart")
+                .build();
+
+        // when
+        Page<Workbook> workbooks = workbookSearchRepository.searchAll(parameter, null, null, parameter.toPageRequest());
+        List<Workbook> workbookList = workbooks.toList();
+
+        // then
+        assertThat(workbookList).hasSize(7);
+        assertThat(workbookList.stream()
+                .map(Workbook::getName)
+                .collect(Collectors.toList())).containsExactly("좋아요가 많아 문제다.",
+                "Java 문제집0",
+                "Javascript 문제집0",
+                "Java 문제집1",
+                "Javascript 문제집1",
+                "Java 문제집2",
+                "Javascript 문제집2");
+    }
+
+    @Test
+    @DisplayName("검색어와 일치해도 카드가 없으면 조회되지 않는다.")
+    void searchAllFromKeywordWhenNotExistCard() {
+        // given
+        WorkbookSearchParameter parameter = WorkbookSearchParameter.builder()
+                .searchKeyword("카드가 없다")
+                .build();
+
+        // when
+        Page<Workbook> workbooks = workbookSearchRepository.searchAll(parameter, null, null, parameter.toPageRequest());
+        List<Workbook> workbookList = workbooks.toList();
+
+        // then
+        assertThat(workbookList).hasSize(0);
```

Comment:
```suggestion
        assertThat(workbookList).isEmpty();
```
isEmpty()로 바꿔도 괜찮을 것 같네요 😁 

SonarQube 나 SonarLint 에서도 잡아주는데, 그 이유가
1. 특정 상황을 위해 제공하는 메서드가 있다면 그것을 활용하자
2. 에러 메시지가 그 상황에 더 어울리게 나온다

라고 합니다. 물론 둘 다 돌려서 에러 메시지 확인해보니까 hasSize를 써도 충분히 이해할 메시지가 나오긴 하더라구요.
그래도 1번 이유로 isEmpty()를 사용하기 적절해 보입니다 😄

Reply:
아하...!! 좋은 정보 감삼다ㅋㅋㅋ 수정했습니다!!!